src/main/kotlin/com/exactpro/th2/ws/client/api/impl/DefaultHandler.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2021-2021 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.ws.client.api.impl
+
+import com.exactpro.th2.ws.client.api.IClient
+import com.exactpro.th2.ws.client.api.IHandler
+import com.exactpro.th2.ws.client.api.IHandlerSettings
+import mu.KotlinLogging
+import java.util.Timer
+import kotlin.concurrent.timer
+
+class DefaultHandler : IHandler {
+    private val logger = KotlinLogging.logger {}
+    private lateinit var settings: DefaultHandlerSettings
+    private lateinit var timer: Timer
+
+    override fun init(settings: IHandlerSettings?) {
+        this.settings = requireNotNull(settings as? DefaultHandlerSettings) {
+            "settings is not an instance of ${DefaultHandlerSettings::class.simpleName}"
+        }
+    }
+
+    override fun onOpen(client: IClient) {
+        createTimer(client)
+    }
+
+    override fun onPing(client: IClient, data: ByteArray) {
+        cancelTimer()
+        createTimer(client)
```

Comment:
Maybe use periodically timer instead

Reply:
This timer is already periodical. I'm cancelling it to avoid sending a ping more often than a set ping interval (i.e. if we have received a ping from a server we have to reset our existing ping-sending-task)